src/libcore/macros.rs

```diff
@@ -2,21 +2,35 @@
 ///
 /// For details, see `std::macros`.
 #[macro_export]
-#[allow_internal_unstable(core_panic)]
+#[allow_internal_unstable(core_panic, panic_internals)]
 #[stable(feature = "core", since = "1.6.0")]
 macro_rules! panic {
     () => (
         $crate::panic!("explicit panic")
     );
     ($msg:expr) => ({
-        $crate::panicking::panic(&($msg, $crate::file!(), $crate::line!(), $crate::column!()))
+        #[cfg(bootstrap)]
+        { $crate::panicking::panic(&($msg, $crate::file!(), $crate::line!(), $crate::column!())) }
+
+        #[cfg(not(bootstrap))]
+        {
```

Comment:
Placing `#[cfg]` inside the expanded code, as opposed to duplicating the macro, can lead to slower expansion of *all* `panic!()` invocations, *and* it would depend on `--cfg bootstrap` *at the use site* (wherever that may be, i.e. in any code compiled with nightly `rustc`), as opposed to while building `libcore`.

Supposedly @Centril asked for this change but I can't find the original comment.

Reply:
https://github.com/rust-lang/rust/pull/65664#discussion_r337051372

Reply:
Ah, I may have misunderstood the request about panicking.rs to also apply to the macros file. Let me know what I should do!